gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/maven/MavenArtifact.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.testing.maven;
+
+import java.util.List;
+import org.immutables.value.Value;
+
+/**
+ * Represents a Maven artifact that can be published to a test repository.
+ * Contains a coordinate and optional dependencies.
+ */
+@Value.Immutable
```

Comment:
Needs `@Value.Style`. You probably want to copy what the internal `immutables-style` repo has.